src/Mollie/WC/Helper/ApplePayDirectHandler.php

```diff
@@ -0,0 +1,1180 @@
+        $applePayRequestDataObject = $this->applePayDataObjectHttp();
+        $applePayRequestDataObject->validationData($_POST);
+
+        if (!$this->isNonceValid($applePayRequestDataObject)) {
+            wp_send_json_error('no nonce');
+        }
+        $validationUrl = $applePayRequestDataObject->validationUrl;
+        $completeDomain = get_site_url();
+        $removeHttp = ["https://", "http://", "/"];
+        $domain = str_replace($removeHttp, "", $completeDomain);
+
+        try {
+            $json = $this->validationApiWalletsEndpointCall(
+                $domain,
+                $validationUrl
+            );
+        } catch (\Mollie\Api\Exceptions\ApiException $exc) {
+            update_option('g_test', $exc->getMessage());
+            update_option('mollie_wc_applepay_validated', 'no');
+            wp_send_json_error($exc->getMessage());
+        }
+
+        update_option('mollie_wc_applepay_validated', 'yes');
+        update_option('mollie_apple_pay_session_response', $json);
+
+        wp_send_json_success($json);
+    }
+
+    /**
+     * Method to validate and update the shipping contact of the user
+     * It updates the amount paying information if needed
+     * On error returns an array of errors to be handled by the script
+     * On success returns the new contact data
+     */
+    public function updateShippingContact()
+    {
+        $applePayRequestDataObject = $this->applePayDataObjectHttp();
+        $applePayRequestDataObject->updateContactData($_POST);
+
+        if (!$this->isNonceValid($applePayRequestDataObject)) {
+            return;
+        }
+        if ($applePayRequestDataObject->hasErrors()) {
+            $this->responseWithDataErrors($applePayRequestDataObject->errors);
+            return;
+        }
+
+        if (class_exists('WC_Countries')) {
+            $countries = $this->createWCCountries();
+            $allowedSellingCountries = $countries->get_allowed_countries();
+            $allowedShippingCountries = $countries->get_shipping_countries();
+            $isAllowedSellingCountry = array_key_exists(
+                $applePayRequestDataObject->simplifiedContact['country'],
+                $allowedSellingCountries
+            );
+
+            $isAllowedShippingCountry = array_key_exists(
+                $applePayRequestDataObject->simplifiedContact['country'],
+                $allowedShippingCountries
+            );
+            $productNeedShipping
+                = $applePayRequestDataObject->needShipping;
+            if (!$isAllowedSellingCountry) {
+                $this->responseWithDataErrors(
+                    [['errorCode' => 'addressUnserviceable']]
+                );
+                return;
+            }
+            if ($productNeedShipping && !$isAllowedShippingCountry) {
+                $this->responseWithDataErrors(
+                    [['errorCode' => 'addressUnserviceable']]
+                );
+                return;
+            }
+        }
```

Comment:
Instead of a big `if` why not return early? The code will be more comprensible and you'll avoid all of those nested ` if` which are difficult to test.